pkg/controller.v1alpha3/suggestion/suggestionclient/suggestionclient.go

```diff
@@ -174,10 +176,92 @@ func (g *General) ConvertExperiment(e *experimentsv1alpha3.Experiment) *suggesti
 }
 
 // ConvertTrials converts CRD to the GRPC definition.
-func (g *General) ConvertTrials(
-	t []trialsv1alpha3.Trial) []*suggestionapi.Trial {
-	res := make([]*suggestionapi.Trial, 0)
-	return res
+func (g *General) ConvertTrials(ts []trialsv1alpha3.Trial) []*suggestionapi.Trial {
```

Comment:
Strange. Without trial information being sent to suggestion algorithm services, how did it work for other algorithms?
@gaocegege

Reply:
We have unit test with trial info. But if the trial info is missing, the algorithm also works. I think it is the reason why we did not find the problem.

Reply:
https://github.com/kubeflow/katib/blob/ebb48f88acb307bcc34f0530a905bcdd456c272f/pkg/suggestion/v1alpha3/skopt_service.py#L30 
 We are using trials from the request. How can bayesianoptimization like algo  work without the previous trial information?

Reply:
bayesianoptimization can work since the prior knowledge is always nil. skopt can handle it.

Reply:
It is strange why we are trying to convert it from `request.trials`. I faced with the problem when `request.trials` is nil, so it is nothing to convert.